src/de/tests.rs

```diff
@@ -44,6 +44,30 @@ fn deserialize_option() {
     assert_eq!(super::from_str("first=23&last=42"), Ok(result));
 }
 
+#[test]
+fn deserialize_option_vec() {
+    #[derive(Deserialize, PartialEq, Debug)]
+    struct Form {
+        value: Option<Vec<String>>,
+    }
+
+    let result = super::from_str("value=abcd&value=def");
+
+    assert_eq!(result, Ok(Form { value: Some(vec!["abcd".to_owned(), "def".to_owned()]) }));
+}
+
+#[test]
+fn deserialize_option_vec_empty() {
+    #[derive(Deserialize, PartialEq, Debug)]
+    struct Form {
+        value: Option<Vec<String>>,
+    }
+
+    let result = super::from_str("");
+
+    assert_eq!(result, Ok(Form { value: None }));
```

Comment:
```suggestion
    assert_eq!(
        super::from_str("value=abc&value=def"),
        Ok(Form { value: Some(vec!["abc".to_owned(), "def".to_owned()]) })
    );
    assert_eq!(super::from_str(""), Ok(Form { value: None }));
```

Reply:
We would have to declare the `Form` outside the test for this. Do you want me to do that?